sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java

```diff
@@ -0,0 +1,135 @@
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.server.resource.authentication.AbstractOAuth2TokenAuthenticationToken;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.MalformedURLException;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class AADOAuth2OboAuthorizedClientRepository implements OAuth2AuthorizedClientRepository {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AADOAuth2OboAuthorizedClientRepository.class);
+
+    private final AzureClientRegistrationRepository azureClientRegistrationRepository;
+
+    private Map<String, ConfidentialClientApplication> confidentialClientApplicationMap = new HashMap<>();
+
+    public AADOAuth2OboAuthorizedClientRepository(AzureClientRegistrationRepository azureClientRegistrationRepository) {
+        this.azureClientRegistrationRepository = azureClientRegistrationRepository;
+        Iterator<ClientRegistration> iterator = azureClientRegistrationRepository.iterator();
+        while (iterator.hasNext()) {
+            ClientRegistration next = iterator.next();
+            this.confidentialClientApplicationMap.put(next.getClientId(), createApp(next));
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends OAuth2AuthorizedClient> T loadAuthorizedClient(String registrationId,
+                                                                     Authentication authentication,
+                                                                     HttpServletRequest request) {
+        try {
+            ClientRegistration clientRegistration = azureClientRegistrationRepository
+                .findByRegistrationId(registrationId);
+
+            AbstractOAuth2TokenAuthenticationToken<AbstractOAuth2Token> authenticationToken =
+                (AbstractOAuth2TokenAuthenticationToken)
+                    authentication;
+
+            String accessToken = authenticationToken.getToken().getTokenValue();
+
+            OnBehalfOfParameters parameters = OnBehalfOfParameters
+                .builder(clientRegistration.getScopes(), new UserAssertion(accessToken))
+                .build();
+
+            ConfidentialClientApplication clientApplication = confidentialClientApplicationMap.get(clientRegistration
+                .getClientId());
+
+            String oboAccessToken = clientApplication.acquireToken(parameters).get().accessToken();
```

Comment:
You're not leveraging the MSAL Token cache here. Every time you need to get a token, you'll make a call to AAD, even if you have a valid cached token. If you want to use the MSAL token cache, we have a [sample](https://github.com/Azure-Samples/ms-identity-java-webapi/blob/master/msal-obo-sample/src/main/java/com/microsoft/azure/msalobosample/MsalAuthHelper.java#L45 ) that shows how you can use the token cache with OBO:

Reply:
We'll not bring the MSAL token cache in this case, since the MSAL token cache is stored in the session. However, we're implementing this OBO function targeting resource servers, which could be stateless and disable session. We'll store the token in the request level, and it will be introduced in another PR.

Reply:
we will create other pr to do it

Reply:
@saragluna Can go a bit more into how you plan on storing the token at the request elvel? Are you storing both the access token and the refresh token?

Reply:
This is how we store the token in request level: 

- Saving the token
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java#L102
- Retrieving the token
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java#L62-L64